tests/run/fors.scala

```diff
@@ -108,6 +110,17 @@ object Test extends App {
     for case (x, y) <- xs do print(s"${(y, x)} "); println()
   }
 
+  /////////////////// elimination of map ///////////////////
+
+  @tailrec
+  def pair[B](xs: List[Int], ys: List[B], n: Int): List[(Int, B)] =
+    if n == 0 then xs.zip(ys)
+    else for (x, y) <- pair(xs.map(_ + 1), ys, n - 1) yield (x, y)
```

Comment:
That doesn't seem to be a compelling case, because I would expect it to be written
```suggestion
    else pair(xs.map(_ + 1), ys, n - 1)
```
to begin with.

Reply:
It's just a test case to verify that it is rewritten to a tail recursive function.